addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -59,17 +63,21 @@ options.registry.SocialMedia = options.Class.extend({
         // Handle element deletation.
         const entriesIds = entries.map(entry => entry.id);
         const anchorsEls = this.$target[0].querySelectorAll(':scope > a');
-        const deletedEl = Array.from(anchorsEls).find(aEl => !entriesIds.includes(aEl.id));
+        const deletedEl = Array.from(anchorsEls)
+            .find(aEl => !entriesIds.includes(this.links.find(linkEl => linkEl.el === aEl).id));
         if (deletedEl) {
             deletedEl.remove();
```

Comment:
You are deleting the node but not the entry in the `this.links leaving dead records inside it (node doesnt exists anymore).
Not sure if that will lead to a bug later, but that's definitely not good